budgetportal/tests/test_prov_infra_projects.py

```diff
@@ -49,92 +61,177 @@ def test_upload_xlsx_for_prov_infra_projects(self):
+    def test_correct_numbers_showed(self):
+        selenium = self.selenium
+        selenium.get("%s%s" % (self.live_server_url, self.url))
+        num_of_projects = selenium.find_element_by_xpath(
+            '//*[@id="num-matching-projects-field"]'
+        ).text
+        num_of_projects = int(num_of_projects)
+        self.assertEqual(num_of_projects, 11)
+
+        num_of_items = len(
+            selenium.find_elements_by_xpath('//*[@id="result-list-container"]/a')
+        )
+        # Header is also counted
+        num_of_items = num_of_items - 1
+        self.assertEqual(num_of_projects, num_of_items)
 
-        # check whether all rows imported (There are 11 rows)
-        count = ProvInfraProject.objects.count()
-        self.assertEqual(count, 11)
+    def test_number_updated_after_search(self):
+        # TODO: Not working yet
+        province = "Eastern Cape"
+        selenium = self.selenium
+        selenium.get("%s%s" % (self.live_server_url, self.url))
+        num_of_projects = selenium.find_element_by_xpath(
+            '//*[@id="num-matching-projects-field"]'
+        ).text
+        num_of_projects = int(num_of_projects)
+        self.assertEqual(num_of_projects, 11)
+
+        search_field = selenium.find_element_by_name("Infrastructure-Search")
+        search_field.send_keys(province)
+        selenium.implicitly_wait(self.timeout)
+        search_button = selenium.find_element_by_id("Search-Button")
+        search_button.click()
+        selenium.implicitly_wait(self.timeout)
+        selenium.refresh()
+        filtered_num_of_projects = selenium.find_element_by_xpath(
+            '//*[@id="num-matching-projects-field"]'
+        ).text
+        filtered_num_of_projects = int(filtered_num_of_projects)
+        self.assertEqual(filtered_num_of_projects, 5)
 
 
-class ProvInfraProjectAPITestCase(APITestCase):
+class ProvInfraProjectAPITestCase(APITransactionTestCase):
     def setUp(self):
         """Create 30 Provincial Infrastructure Projects"""
 
         self.fin_year = FinancialYear.objects.create(slug="2030-31")
-        self.url = reverse("provincial-infrastructure-project-api")
+        self.quarter = Quarter.objects.create(number=1)
+        self.irm_snapshot = IRMSnapshot.objects.create(
+            financial_year=self.fin_year,
+            quarter=self.quarter,
+            date_taken=date(year=2019, month=1, day=1),
+        )
+        self.url = reverse("provincial-infrastructure-project-api-list")
+        self.facet_url = reverse("provincial-infrastructure-project-api-facets")
         self.provinces = ["Eastern Cape", "Free State"]
         self.statuses = ["Design", "Construction"]
         self.sources = [
             "Education Infrastructure Grant",
             "Community Library Service Grant",
         ]
+
         for i in range(30):
             if i < 15:
-                status = self.statuses[0]
+                status_ = self.statuses[0]
```

Comment:
It was confused with _rest_framework.status_ . That's why I added underscore

Reply:
Ah I see - I think a better solution is to import the status codes themselves - since the word status is quite a generic word to import by itself in the entire module. Then you don't have a conflict like this